src/prommis/uky/costing/tests/test_ree_costing.py

```diff
@@ -4434,3 +4465,98 @@ def test_REE_costing_recovery_Nonewithtransportcost():
             ],
             transport_cost_per_ton_product=10,
         )
+
+
+@pytest.mark.component
+def test_REE_costing_NPV_defaults():
```

Comment:
Building on all of this, you have a lot of `if/else` branches and possible exception paths in the code, but you are not testing any of these here. You need to have a lot more unit testing of these different paths, especially given how complex the code for this model is.